codec.go

```diff
@@ -49,6 +51,30 @@ type Codec interface {
 	Unmarshal([]byte, any) error
 }
 
+// StableCodec is an extension to Codec for serializing with stable output.
+type StableCodec interface {
+	// MarshalStable marshals the given message with stable field ordering.
+	//
+	// MarshalStable should return the same output for a given input. Although
+	// it is not guaranteed to be canonicalized, the marshalling routine for
+	// MarshalStable will opt for the most normalized output available for a
+	// given serialization.
+	//
+	// For practical reasons, it is possible for MarshalStable to return two
+	// different results for two inputs considered to be "equal" in their own
+	// domain, and it may change in the future with codec updates, but for
+	// any given concrete value and any given version, it should return the
+	// same output.
+	MarshalStable(any) ([]byte, error)
```

Comment:
Where is this called?

Reply:
Not yet. Note that this PR merges into a feature branch, not `main`, with the intent of being able to have code reviewed before incrementally before it's ready. I would like to add tests for this, but I suspect the choice to not test the `codec.go` code directly (thusfar) was deliberate.